src/app/components/Ad/index.tsx

```diff
@@ -1,13 +1,23 @@
 import React, { useContext } from 'react';
 import useToggle from '#hooks/useToggle';
 import { RequestContext } from '#contexts/RequestContext';
+import crypto from 'crypto';
 import AmpAd from './Amp';
 import CanonicalAd from './Canonical';
 import { AdProps } from './types';
 import AdBootstrapJs from './Canonical/AdBootstrapJs';
 
 const AdContainer = ({ slotType, className, adcampaign }: AdProps) => {
-  const { isAmp, isLite, showAdsBasedOnLocation } = useContext(RequestContext);
+  const { isAmp, isLite, showAdsBasedOnLocation, country } =
+    useContext(RequestContext);
+  const { enabled: isAdsNonceEnabled, value: countries } =
+    useToggle('adsNonce');
+
+  const isNonceAllowed = isAdsNonceEnabled && countries.includes(country);
```

Comment:
I think we should convert the `countries` string into an array. Also, we should cater for the future scenario where we don't configure any countries in the toggle, meaning that it should be enabled for all countries. 

Something like this:
```suggestion
  const countriesForNonce = countries?.split(',');
  const nonceEnabledForCountry = countriesForNonce.length === 0 || countriesForNonce.includes(country);
  const isNonceAllowed = isAdsNonceEnabled && nonceEnabledForCountry;
```